engine/src/test/java/io/camunda/zeebe/streamprocessor/StreamPlatform.java

```diff
@@ -117,19 +115,23 @@ public CommandResponseWriter getMockCommandResponseWriter() {
   }
 
   public void createLogStream() {
-    final var logStorage = new ListLogStorage();
+    createLogStream(new ListLogStorage(), DEFAULT_PARTITION);
+  }
+
+  public LogContext createLogStream(final ListLogStorage logStorage, final int partitionId) {
     final var logStream =
         SyncLogStream.builder()
-            .withLogName(STREAM_NAME + DEFAULT_PARTITION)
+            .withLogName(STREAM_NAME + partitionId)
             .withLogStorage(logStorage)
-            .withPartitionId(DEFAULT_PARTITION)
+            .withPartitionId(partitionId)
             .withActorSchedulingService(actorScheduler)
             .build();
 
     logStorage.setPositionListener(logStream::setLastWrittenPosition);
 
     logContext = new LogContext(logStream);
```

Comment:
:thought_balloon: So if I call `createLogStream(logStorage, partitionId)` I override the current `logContext`. On `startStreamProcessor`, the stream processor is built with the logstream from the context that was last created. That sounds like an easy trap to fall into. I'm not sure if it would cause problems and how easy they would be to spot.

Could/Should we prevent this by not setting `logContext` here and requiring a call to `buildStreamProcessor` instead of `startStreamProcessor` to start the stream processor in that case?

Reply:
Yeah good point. Tbh I wouldn't expect that you use that method normally, but I agree we should try to prevent such issues :)

Reply:
>  I'm not sure if it would cause problems and how easy they would be to spot.

> Could/Should we prevent this by not setting logContext here and requiring a call to buildStreamProcessor instead of startStreamProcessor to start the stream processor in that case?

just want to add something here. The logContext will always also be closed due to addition to the closable list and it always used a new listLogStorage so there shouldn't be any issues in general.

But I will try to improve the interface here :)